x-pack/plugins/task_manager/server/task_running/task_runner.ts

```diff
@@ -514,20 +524,28 @@ export class TaskManagerRunner implements TaskRunner {
       unwrap
     )(result);
 
-    this.instance = asRan(
-      await this.bufferedTaskStore.update(
-        defaults(
-          {
-            ...fieldUpdates,
-            // reset fields that track the lifecycle of the concluded `task run`
-            startedAt: null,
-            retryAt: null,
-            ownerId: null,
-          },
-          this.instance.task
+    if (!this.isExpired) {
+      this.instance = asRan(
+        await this.bufferedTaskStore.update(
+          defaults(
+            {
+              ...fieldUpdates,
+              // reset fields that track the lifecycle of the concluded `task run`
+              startedAt: null,
+              retryAt: null,
+              ownerId: null,
+            },
+            this.instance.task
+          )
         )
-      )
-    );
+      );
+    } else {
```

Comment:
Do you think it'd be useful to try to log a message including the timeout value and the actual task execution duration?

Reply:
I'm actually not sure. This change isn't really something users need to know about is it? I'm not even sure what it'd say as we're just not updating the rule SO which feels like a detail users won't understand. 

We already have server logs for when a task expires so the user will know something happened.

WDYT?

Reply:
I'm on the fence. I think we probably don't need it for alerting tasks since we have the event log but could it be useful for other task types that are running long? I think we can leave it for now and if it's needed later we can add it in.

Reply:
Gotcha, I'm inclined to leave it as is for now and we can readdress if needed in the future